### ISSUE TITLE
Dynamic Reference Variable Slot does not Clear Null References on Delete of the Slot

### ISSUE THREAD

**Issue by Ardescar** [Neos-Metaverse/NeosPublic#2867]
### Describe the bug?

When i delete a slot then the dynamic variable reference does Not get nullified. (see screenshot)

It only gets nullified later when i grab the box - or do other actions with the object that refreshes it.

### Relevant issues

__I did not really find other issues.__
After creating Issue this came up https://github.com/Neos-Metaverse/NeosPublic/issues/1394

### To Reproduce

Just like in my screenshots 

- create a DynRefVar <Slot>
- put the Reference in there 
- then delete the Slot

### Expected behavior

that the reference entry does get removed when the slot gets killed

### Screenshots

The Setup.
![Setup](https://user-images.githubusercontent.com/63201013/131114665-397057c1-96b4-4061-be3b-42e8c5fca3f1.jpg)
After deleting the Box.
![afterdeleting](https://user-images.githubusercontent.com/63201013/131114787-8c77cadb-cb9b-4200-9eb4-86ca554d9d88.jpg)
After just Grabbing the Box where all is in.
![aftergrabbingthebox](https://user-images.githubusercontent.com/63201013/131114849-1ca76e50-ec51-4e5d-9a29-ccc5c8db0a15.jpg)

### How often does it happen?

Always

### Does the bug persist after restarting Neos?

Yes

### Neos Version Number

2021.8.27.199 (did not work for weeks)

### What Platforms does this occur on?

Windows

### Did this work before?

No

### Reporters

Ardes - Ardescar#2046

**Comment by Earthmark** [comment]
For some context I believe this happens with the input node variant as well, I had this issue come up yesterday. However I'll need to verify that!

**Comment by H3BO3** [comment]
#1394 ?

**Comment by Ardescar** [comment]
@h3bo3 looks like partially the same issue. just maybe because of changes in neos its now different?
I mean as soon as it shows NULL inside a inspector the Dynamic Vars are fine again. But is it a UI refresh?
I mean i have to do something with the objekt - like grabbing it - that it refreshes.

**Comment by Earthmark** [comment]
I agree it looks related, I had a related issue with a dynamic variable name being driven based on it's parent's name, the value would be stale unless you inspected the slot with an inspector or duplicated the slot, at which point the new version would connect and the old one would disconnect.

A few of my things that use drive based dyn-var-naming have to do the rename copy thing.

I think some pull on the value, either populating the inspector with text values, or a slot getting patented (not sure why this last one would do it but hey) cause the values to get reevaluated and the changes show up.

Based on a conversation in #2863 this may be an evaluation trigger thing instead of a cache and hold thing.
